sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -91,8 +91,18 @@ private[sql] class HiveSessionCatalog(
               isUDAFBridgeRequired = true))
             udfExpr.get.dataType // Force it to check input data types.
           } else if (classOf[GenericUDTF].isAssignableFrom(clazz)) {
-            udfExpr = Some(HiveGenericUDTF(name, new HiveFunctionWrapper(clazz.getName), input))
-            udfExpr.get.asInstanceOf[HiveGenericUDTF].elementSchema // Force it to check data types.
+            try {
+              udfExpr = Some(HiveGenericUDTF(name, new HiveFunctionWrapper(clazz.getName), input))
+              // Force it to check input data types.
+              udfExpr.get.asInstanceOf[HiveGenericUDTF].elementSchema
+            } catch {
+              case exception: Exception =>
```

Comment:
why do we need this fallback mechanism? can we just switch to the new API without the `deprecated` flag?